lib/improver/psychrometric_calculations/psychrometric_calculations.py

```diff
@@ -612,53 +593,57 @@ def process(self, temperature, relative_humidity, pressure):
         units_string = "K {}".format(
             wet_bulb_temperature.coord(self.coord_name_to_integrate).units)
         wet_bulb_temperature_integral.units = Unit(units_string)
-        return wet_bulb_temperature, wet_bulb_temperature_integral
+        return wet_bulb_temperature_integral
 
 
-class FallingSnowLevel(BasePlugin):
-    """Calculate a field of continuous falling snow level."""
+class PhaseChangeLevel(BasePlugin):
+    """Calculate a continuous field of heights relative to sea level at which
+    a phase change of precipitation is expected."""
 
-    def __init__(self, precision=0.005, falling_level_threshold=90.0,
-                 grid_point_radius=2):
+    def __init__(self, phase_change, grid_point_radius=2):
         """
         Initialise class.
 
         Args:
-            precision (float):
-                The precision to which the Newton iterator must converge
-                before returning wet bulb temperatures.
-            falling_level_threshold (float):
-                The cutoff threshold for the Wet-bulb integral used
-                to calculate the falling snow level.We are integrating to the
-                threshold that is presumed to indicate
-                the level at which snow has melted back to rain. Above
-                this level we should have falling snow.
+            phase_change (str):
+                The desired phase change for which the altitude should be
+                returned. Options are:
+
+                    snow-sleet - the melting of snow to sleet.
+                    sleet-rain - the melting of sleet to rain.
+
             grid_point_radius (int):
                 The radius in grid points used to calculate the maximum
                 height of the orography in a neighbourhood as part of this
                 calculation.
-
         """
-        self.precision = precision
-        self.wet_bulb_integral_plugin = (
-            WetBulbTemperatureIntegral(precision=precision))
-        self.falling_level_threshold = falling_level_threshold
+        phase_changes = {'snow-sleet': {'threshold': 90.,
+                                        'name': 'snow_falling'},
+                         'sleet-rain': {'threshold': 202.5,
+                                        'name': 'rain_falling'}}
+        try:
+            phase_change_def = phase_changes[phase_change]
+        except KeyError:
+            msg = ("Unknown phase change '{}' requested.\nAvailable options "
+                   "are: {}".format(phase_change, phase_changes.keys()))
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [dict.keys referenced when not iterating](https://app.codacy.com/app/metoppv_tech/improver/pullRequest?prid=4624317)